tiny_dnn/layers/convolutional_layer.h

```diff
@@ -232,13 +233,16 @@ class convolutional_layer : public feedforward_layer<Activation> {
      **/
     void forward_propagation(const std::vector<tensor_t*>& in_data,
                              std::vector<tensor_t*>&       out_data) override { 
-        std::cout << "BEFORE PADDED" << std::endl;
         copy_and_pad_input(*in_data[0], cws_.prev_out_padded_);
-        *in_data[0] = cws_.prev_out_padded_;
-        std::cout << "AFTER PADDED" << std::endl;
+
+        std::vector<tensor_t*> in_data_;
+        in_data_.push_back(&cws_.prev_out_padded_);
+        for (cnn_size_t i = 1; i < in_data.size(); ++i) {
+            in_data_.push_back(in_data[i]);
+        }
 
```

Comment:
@nyanp I've tried this workaround solution but I think we are doing sth wrong since all tests for conv are broken in this way :disappointed: